test/openAPI/features/client_update.feature

```diff
@@ -1,16 +1,110 @@
+@method=PUT @endpoint=/client-mgmt/oidc-client/{client_id}
 Feature: API to update existing Open ID Connect (OIDC) client.
 
-  Update OIDC Client Endpoint.
+  @smoke @unit @positive
+  Scenario: The client profile is successfully updated in the Open ID Connect (OIDC) smoke type test
+    Given The user wants to update the client profile in the Open ID Connect (OIDC)
+    And The client profile with "e-health-service" as clientId is created
+    When User sends PUT request with given "e-health-service" as client_id parameter, "2011-10-05T14:48:00.000Z" as requestTime, "Health Service" as clientName, "active" as status, "http://example.com" as logoUri, "http://example-redirect.com" as redirectUris, "name" as userClaims, "idbb:acr:generated-code" as authContextRefs, "authorization_code" as grantTypes, "private_key_jwt" as clientAuthMethods
+    Then User receives a response from the PUT /client-mgmt/oidc-client/{client_id} endpoint
+    And The PUT /client-mgmt/oidc-client/{client_id} endpoint response should be returned in a timely manner 15000ms
+    And The PUT /client-mgmt/oidc-client/{client_id} endpoint response should have status 200
+    And The PUT /client-mgmt/oidc-client/{client_id} endpoint response should have content-type: application/json header
+    And The PUT /client-mgmt/oidc-client/{client_id} endpoint response should match json schema
+    And The PUT /client-mgmt/oidc-client/{client_id} endpoint response should contain "e-health-service" as clientId
 
-  Request endpoint: PUT /client-mgmt/oidc-client/{client_id}
+  @unit @negative
+  Scenario: Not able to update the client because of invalid clientAuthMethods
+    Given The user wants to add the new client to the Open ID Connect (OIDC)
+    And The client profile with "e-health-service-01" as clientId is created
+    When User sends PUT request with given "invalid_auth_method" as invalid clientAuthMethods
+    Then User receives a response from the PUT /client-mgmt/oidc-client/{client_id} endpoint
+    And The PUT /client-mgmt/oidc-client/{client_id} endpoint response should be returned in a timely manner 15000ms
+    And The PUT /client-mgmt/oidc-client/{client_id} endpoint response should have status 200
+    And The PUT /client-mgmt/oidc-client/{client_id} endpoint response should have content-type: application/json header
+    And The PUT /client-mgmt/oidc-client/{client_id} endpoint response should match json schema with error
```

Comment:
response is not a required object, so json schema match should still be successfull for all the negative scenarios

Reply:
It is. In this test case I check if in the response errors is not an empty array because it schouldn't

Reply:
It means that in the response error field is not empty, there should be error but maybe this step is not necessary because below I check if there is specific errorCode. I will remove it then

Reply:
If we just match the response with open API schema for the response, we can still keep it since response is not a required field

Reply:
Ok, so I can leave it, right?